python/ray/air/predictor.py

```diff
@@ -32,17 +70,45 @@ def from_checkpoint(cls, checkpoint: Checkpoint, **kwargs) -> "Predictor":
         """
         raise NotImplementedError
 
+    @PublicAPI(stability="alpha")
     def predict(self, data: DataBatchType, **kwargs) -> DataBatchType:
         """Perform inference on a batch of data.
 
         Args:
-            data: A batch of input data. Either a pandas Dataframe or numpy
-                array.
-            kwargs: Arguments specific to predictor implementations.
+            data: A batch of input data of type ``DataBatchType``.
+            kwargs: Arguments specific to predictor implementations. These are passed
+            directly to ``_predict_pandas``.
 
         Returns:
             DataBatchType: Prediction result.
         """
+
+        pass
+
+        # TODO(amogkam): Implement below code.
+        # data_df = _convert_batch_type_to_pandas(data)
+        #
+        # if hasattr(self, "preprocessor") and self.preprocessor:
+        #     data_df = self.preprocessor.transform_batch(data_df)
+        #
+        # predictions_df = self._predict_pandas(data_df)
+        # return _convert_pandas_to_batch_type(predictions_df, type=type(data))
+
+    @DeveloperAPI
+    @abc.abstractmethod
```

Comment:
Let's not make it abstractmethod, since subclasses can choose to override predictor instead. Just have it raise NotImplementedError by default.

Reply:
ok yep good point, updated!